GKE-runner/main.tf

```diff
@@ -0,0 +1,187 @@
+  bucket_expiration_days = var.cache_expiration_days
+  prefix                 = var.prefix
+}
+
+
+#--------------------------------------------------------
+# create kubernetes secret from service account cred file
+#--------------------------------------------------------
+
+resource "google_service_account_key" "cache_admin" {
+  count              = local.count
+  service_account_id = module.cache.0.cache_service_account_name
+}
+
+
+resource "kubernetes_namespace" "runner_namespace" {
+  metadata {
+    name = var.namespace
+  }
+
+  depends_on = [
+    google_container_node_pool.gitlab_runner_pool
+  ]
+}
+
+resource "kubernetes_secret" "cache_secret" {
+  count = local.count
+  metadata {
+    name      = local.cache_secret_name
+    namespace = kubernetes_namespace.runner_namespace.metadata[0].name
+  }
+
+  binary_data = {
+    gcs_cred = google_service_account_key.cache_admin[0].private_key
+  }
+
+  depends_on = [
+    google_container_node_pool.gitlab_runner_pool,
+    kubernetes_namespace.runner_namespace
+  ]
+}
+
+
+#----------------------------------------------------------------------
+# set up gitlab runner using the deimos kubernetes gitlab runner module
+#-----------------------------------------------------------------------
+module "kubernetes_gitlab_runner" {
+  source  = "DeimosCloud/gitlab-runner/kubernetes"
+  version = "~>1.3.0"
+
+  release_name  = local.release
+  chart_version = var.chart_version
+  namespace     = var.namespace
+
+  gitlab_url = var.gitlab_url
+  concurrent = var.concurrent
+  replicas   = var.replicas
+
+  runner_name               = local.runner_name
+  runner_token              = var.runner_token
+  runner_tags               = var.runner_tags
+  runner_registration_token = var.runner_registration_token
+  runner_locked             = var.runner_locked
+  runner_image              = var.runner_image
+  run_untagged_jobs         = var.run_untagged_jobs
+  unregister_runners        = var.unregister_runners
+
+  manager_node_selectors   = var.runner_node_pool_node_labels
+  manager_node_tolerations = var.manager_node_tolerations
+  manager_pod_annotations  = var.manager_pod_annotations
+  manager_pod_labels       = var.manager_pod_labels
+
+  build_job_node_selectors        = var.build_job_node_selectors
+  build_job_node_tolerations      = var.build_job_node_tolerations
+  build_job_secret_volumes        = var.build_job_secret_volumes
+  build_job_mount_docker_socket   = var.build_job_mount_docker_socket
+  build_job_run_container_as_user = var.build_job_run_container_as_user
+
+  docker_fs_group = var.docker_fs_group
+
+  image_pull_secrets                 = var.image_pull_secrets
+  create_service_account             = var.runner_create_service_account
+  service_account_clusterwide_access = var.runner_service_account_clusterwide_access
+
+  cache = {
+    type   = var.cache_type
+    path   = var.cache_path
+    shared = var.cache_shared
+    gcs    = local.gcs
```

Comment:
If we only set this, what other values can `var.cache_type` have? we can just set the `type` to a `local.cache_type` with value is set to `gcs`

Reply:
I dont think i understand this question. but here, cache type could either be "local" or "gcs" or even "s3" or "azure" in multi cloud setups. the idea behind local.gcs is to set the credentials if the cache type is gcs  and otherwise set it to an empty map

Reply:
This is a google gitlab runner deployment. That means all the resources are to be provisioned on gcp.

Reply:
what about the local cache situation?